### PR TITLE
Assert correct class of arrays returned by ivy and frontend functions/methods

### PR DESCRIPTION
**Pull request by AnnaTz**

### CODE REVIEW

**Comment by ivy-seed** [comment]
This PR has been labelled as stale because it has been inactive for more than 7 days. If you would like to continue working on this PR, then please add another comment or this PR will be closed in 7 days.

**Comment by AnnaTz** [comment]
ivy-gardener

File: ivy_tests/test_ivy/helpers/function_testing.py [hunk 1]
```diff
@@ -280,6 +280,10 @@ def test_function(
             **kwargs,
         )
 
+        assert ivy.nested_map(
+            ret_from_target, lambda x: ivy.is_ivy_array(x) if ivy.is_array(x) else True
+        ), "Ivy function returned non-ivy arrays: {}".format(ret_from_target)
+
```
Comment (CatB1t) [inline ivy_tests/test_ivy/helpers/function_testing.py:286]: We should do similar assertion for return from ground truth.

File: ivy_tests/test_ivy/helpers/function_testing.py [hunk 2]
```diff
@@ -1223,6 +1231,10 @@ def test_method(
         else:
             ret_device = None
 
+        assert ivy.nested_map(
+            ret, lambda x: ivy.is_ivy_array(x) if ivy.is_array(x) else True
+        ), "Ivy method returned non-ivy arrays: {}".format(ret)
+
```
Comment (CatB1t) [inline ivy_tests/test_ivy/helpers/function_testing.py:1237]: Similar to the case above, we should assert the same for ground truth backend.

**Review by CatB1t — CHANGES_REQUESTED** [review]

File: ivy_tests/test_ivy/helpers/function_testing.py [hunk 3]
```diff
@@ -382,6 +386,9 @@ def test_function(
             test_compile=test_flags.test_compile,
             **kwargs,
         )
+        assert ivy.nested_map(
+            ret_from_target, lambda x: ivy.is_ivy_array(x) if ivy.is_array(x) else True
```
Comment (CatB1t) [inline ivy_tests/test_ivy/helpers/function_testing.py:390]: This should be `ret_from_gt`

Comment (CatB1t) [inline ivy_tests/test_ivy/helpers/function_testing.py:390]: We shouldn't use the global ivy, we should use the local version from the context manager `gt_backend`

File: ivy_tests/test_ivy/helpers/function_testing.py [hunk 4]
```diff
@@ -1266,7 +1281,10 @@ def test_method(
             test_compile=test_compile,
             **kwargs_gt_method,
         )
-
+        assert ivy.nested_map(
+            ret, lambda x: ivy.is_ivy_array(x) if ivy.is_array(x) else True
+        ), "Ground-truth method returned non-ivy arrays: {}".format(ret_from_gt)
+
```
Comment (CatB1t) [inline ivy_tests/test_ivy/helpers/function_testing.py:1287]: similarly we should use the local backend

**Review by CatB1t — CHANGES_REQUESTED** [review]

File: ivy_tests/test_ivy/helpers/function_testing.py [hunk 5]
```diff
@@ -280,6 +280,10 @@ def test_function(
             **kwargs,
         )
 
+        assert ivy.nested_map(
+            ret_from_target, lambda x: ivy.is_ivy_array(x) if ivy.is_array(x) else True
```
Comment (CatB1t) [inline ivy_tests/test_ivy/helpers/function_testing.py:284]: `ivy.is_ivy_array()` would be bound to global ivy, which has it's backend unchanged, we should use local Ivy.

File: ivy_tests/test_ivy/helpers/function_testing.py [hunk 6]
```diff
@@ -615,6 +622,10 @@ def test_frontend_function(
             **kwargs_for_test,
         )
 
+        assert ivy.nested_map(
+            ret, lambda x: _is_frontend_array(x) if ivy.is_array(x) else True
```
Comment (CatB1t) [inline ivy_tests/test_ivy/helpers/function_testing.py:626]: `ivy.is_array()` would be bound to global ivy, which has it's backend unchanged, we should use local Ivy.

File: ivy_tests/test_ivy/helpers/function_testing.py [hunk 7]
```diff
@@ -1223,6 +1234,10 @@ def test_method(
         else:
             ret_device = None
 
+        assert ivy.nested_map(
+            ret, lambda x: ivy.is_ivy_array(x) if ivy.is_array(x) else True
```
Comment (CatB1t) [inline ivy_tests/test_ivy/helpers/function_testing.py:1238]: `ivy.is_ivy_array()` would be bound to global ivy, which has it's backend unchanged, we should use local Ivy.

File: ivy_tests/test_ivy/helpers/function_testing.py [hunk 8]
```diff
@@ -1584,6 +1602,10 @@ def test_frontend_method(
             **kwargs_method,
         )
 
+        assert ivy.nested_map(
+            ret, lambda x: _is_frontend_array(x) if ivy.is_array(x) else True
```
Comment (CatB1t) [inline ivy_tests/test_ivy/helpers/function_testing.py:1606]: `ivy.is_array()` would be bound to global ivy, which has it's backend unchanged, we should use local Ivy.

**Review by CatB1t — CHANGES_REQUESTED** [review]

**Comment by ivy-seed** [comment]
This PR has been labelled as stale because it has been inactive for more than 7 days. If you would like to continue working on this PR, then please add another comment or this PR will be closed in 7 days.

**Review by CatB1t — APPROVED** [review]
lgtm!

**Pull request merged**